pkg/upgrade/plan/plan.go

```diff
@@ -125,13 +124,20 @@ func RegisterHandlers(ctx context.Context, serviceAccountName, controllerNamespa
 				defer plans.Enqueue(obj.Namespace, planName)
 				if obj.Status.Succeeded == 1 {
 					planLabel := upgradeapi.LabelPlanName(planName)
-					if version, ok := obj.Labels[planLabel]; ok {
+					if planHash, ok := obj.Labels[planLabel]; ok {
 						if nodeName, ok := obj.Labels[upgradeapi.LabelNode]; ok {
 							node, err := nodes.Get(nodeName, metav1.GetOptions{})
```

Comment:
just a nit: shouldn't we use a node cache to be quicker on a large scale here?

Reply:
@galal-hussein plan cache too then I would think.

Reply:
pushed new commit that uses caches